### PR TITLE
raising exception for improper use of addtoblock

### PR DESCRIPTION
**Pull request by calellowitz**
@dannyroberts 
Added the hard failure when addtoblock is called too late. Is this what you had in mind?
cc: @nickpell

### CODE REVIEW

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 1]
```diff
@@ -444,7 +445,10 @@ def __init__(self, nodelist, block_name):
         self.block_name = block_name
 
     def write(self, context, text):
-        request_blocks = self.get_request_blocks(context)
+        rendered_blocks = AppendingBlockNode.get_rendered_blocks_dict(context)
+        if self.block_name in rendered_blocks:
+            raise AlreadyRenderedException('Block {} already rendered. Cannot add new node'.format(self.block_name))
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: E501 line too long (116 > 115 characters)

**Comment by czue** [comment]
build fail looks real (maybe the exact issue you were looking for?)

```

======================================================================
ERROR: test_status_codes (corehq.apps.prelogin.tests.test_views.TestViews)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/prelogin/tests/test_views.py", line 15, in test_status_codes
    response = self.client.get(reverse(view.urlname), follow=False)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 470, in get
    **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 286, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 358, in generic
    return self.request(**r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 440, in request
    six.reraise(*exc_info)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 137, in get_response
    response = response.render()
  File "/vendor/lib/python2.7/site-packages/django/template/response.py", line 103, in render
    self.content = self.rendered_content
  File "/vendor/lib/python2.7/site-packages/django/template/response.py", line 80, in rendered_content
    content = template.render(context)
  File "/vendor/lib/python2.7/site-packages/django/template/base.py", line 148, in render
    return self._render(context)
  File "/vendor/lib/python2.7/site-packages/django/test/utils.py", line 88, in instrumented_test_render
    return self.nodelist.render(context)
  File "/vendor/lib/python2.7/site-packages/django/template/base.py", line 844, in render
    bit = self.render_node(node, context)
  File "/vendor/lib/python2.7/site-packages/django/template/debug.py", line 80, in render_node
    return node.render(context)
  File "/vendor/lib/python2.7/site-packages/django/template/loader_tags.py", line 126, in render
    return compiled_parent._render(context)
  File "/vendor/lib/python2.7/site-packages/django/test/utils.py", line 88, in instrumented_test_render
    return self.nodelist.render(context)
  File "/vendor/lib/python2.7/site-packages/django/template/base.py", line 844, in render
    bit = self.render_node(node, context)
  File "/vendor/lib/python2.7/site-packages/django/template/debug.py", line 80, in render_node
    return node.render(context)
  File "/vendor/lib/python2.7/site-packages/django/template/loader_tags.py", line 150, in render
    return template.render(context)
  File "/vendor/lib/python2.7/site-packages/django/template/base.py", line 148, in render
    return self._render(context)
  File "/vendor/lib/python2.7/site-packages/django/test/utils.py", line 88, in instrumented_test_render
    return self.nodelist.render(context)
  File "/vendor/lib/python2.7/site-packages/django/template/base.py", line 844, in render
    bit = self.render_node(node, context)
  File "/vendor/lib/python2.7/site-packages/django/template/debug.py", line 80, in render_node
    return node.render(context)
  File "/mnt/commcare-hq-ro/corehq/apps/hqwebapp/templatetags/hq_shared_tags.py", line 458, in render
    self.write(context, output)
  File "/mnt/commcare-hq-ro/corehq/apps/hqwebapp/templatetags/hq_shared_tags.py", line 450, in write
    raise AlreadyRenderedException('Block {} already rendered. Cannot add new node'.format(self.block_name))
AlreadyRenderedException: Block js-inline already rendered. Cannot add new node
-------------------- >> begin captured logging << --------------------
```

**Comment by NoahCarnahan** [comment]
That error should be fixed here! https://github.com/dimagi/commcarehq-prelogin/pull/93
I'm glad these won't fail silently anymore :)

**Comment by calellowitz** [comment]
waiting for the submodule update that fixes this to make it to master, then i will merge master to kick off the tests again. just want to make sure it passes tests first

**Comment by calellowitz** [comment]
@dannyroberts I have implemented a version of what we discussed that relies overriding include blocks. Curious if that solution makes sense to you. Tested that it previously broken pages (like the report ones) work as expected. I plan on throwing this up on staging to make sure no other pages are secretly broken, but should be good to go.

**Comment by dannyroberts** [comment]
Sorry for the delay. Just want to check that you're sure the include-counting method you chose works; I don't have a deep enough knowledge of the system to know off-hand. If it's on staging and you've clicked around the main pages in reports and app builder, I'd say I'm happy with this approach!

**Comment by calellowitz** [comment]
@dannyroberts It has been on staging for a few days, and I clicked around a bit (both to reports that had been broken before and the app builder), and have not seen any problems. Additionally there have not been any 500 emails with that exception, so it seems good to go.

**Pull request merged**